### PR TITLE
#3462 [Bug Report] Total matched invoice qty https://github.com/ademp…

### PR DESCRIPTION
**Pull request by e-Evolution**
…iere/adempiere/issues/3462

### CODE REVIEW

File: base/src/org/compiere/process/InvoiceCreateFrom.java [hunk 1]
```diff
@@ -112,22 +112,7 @@ protected String doIt() throws Exception {
 				MOrderLine orderLine = new MOrderLine (getCtx(), key, get_TrxName());
 				//	Set reference
 				referenceId.set(orderLine.getC_Order_ID());
-				//	Set InOut
-				String whereClause = "EXISTS (SELECT 1 "
-						+ "FROM M_InOut io "
-						+ "WHERE io.M_InOut_ID = M_InOutLine.M_InOut_ID "
-						+ "AND io.DocStatus IN ('CO','CL'))";
-				MInOutLine[] inOutLines = MInOutLine.getOfOrderLine(Env.getCtx(), key, whereClause, get_TrxName());
-				log.fine ("Receipt Lines with OrderLine = #" + inOutLines.length);
```
Comment (yamelsenih) [inline base/src/org/compiere/process/InvoiceCreateFrom.java:121]: This code allows allocate receipt from order when a order have receipt allocated.

How to think you resolve this?

**Review by yamelsenih — CHANGES_REQUESTED** [review]

**Comment by yamelsenih** [comment]
Hi @e-Evolution I think that this change make a bad implemetation for core, the original funcionality allows allocate the receipt with invoice and is a legacy funcionality.

Maybe is better if the original bug is resolved. I can make a local environment for test it tomorrow.

Best regards.

File: base/src/org/compiere/process/InvoiceCreateFrom.java (hunk 1 above)
Comment (e-Evolution) [reply base/src/org/compiere/process/InvoiceCreateFrom.java:121]: The invoice match with purchase order was designed to validate the invoiced quantity with respect to the ordered quantity, so it should not be associated with a material receipt, for that it is the correspondence of receipt of materials with purchase order.

So these lines of code cause bad behavior.

The logic assigns the first receipt of materials that it finds even when it already has a previous match, in the case of partial or multiple receipts it generates the match for the same receipt of material causing a big problem with the matches, having to fix this manually which implies time and confusion in users

Comment (yamelsenih) [reply base/src/org/compiere/process/InvoiceCreateFrom.java:121]: Please put here a step by step for replicate it. This funcionality is legacy and has very little changes. I want to be sure of it

**Comment by yamelsenih** [comment]
Hi @e-Evolution Please put here a step by step for replicate it. This funcionality is legacy and has very little changes. I want to be sure of it

**Comment by marcalwestf** [comment]
@yamelsenih  Yamel, @e-Evolution  Victor provided the step-by-step reproduction instructions with issue https://github.com/adempiere/adempiere/issues/3462

**Comment by marcalwestf** [comment]
@e-Evolution, @yamelsenih  as I understand this pull request, the error occurs when you:
1.- Complete a Purchase Order.
2.- Create an Invoice (Vendor) and use the Purchase Order to generate an Invoice Line, but not for the whole quantity ordered. 
3.- Complete the Invoice (Vendor).
4.- Stay in the Invoice (Vendor) window and create a Material Receipt. Complete the Material Receipt.
5.- Create another Invoice (Vendor) and use the Purchase Order to get an Invoice Line, but not for the whole quantity ordered. 

-> Here you can see that the Material Receipt Line of step 4 is referenced in the Invoice Line of the second Invoice, which is an error because there has not been a Material Receipt for this Invoice.

This behavior can be reproduced with standard Adempiere and it is suppressed with this Pull request.

**Comment by marcalwestf** [comment]
...the deleted code that @yamelsenih asks to be explained is in my opinion dangerous because it gets the first material receipt line that matches the invoice quantity or the first material receipt line if none matches. 

This means if there is already one material receipt line and the quantity doesn't match, it will be referenced. And this is an error.
So in my opinion is OK to eliminate the reference to MInOutLine.

**Review by marcalwestf — APPROVED** [review]
- error was reproduced on standard Adempiere
- error was fixed with changes by this pull request

-> Test passed

**Comment by marcalwestf** [comment]
@e-Evolution  Victor, there are some checks pending
Thanks

@yamelsenih perhaps you can review also the changes and cast your opinion.

**Comment by yamelsenih** [comment]
I agree, just update this branch with last changes for approval from CI

**Review by yamelsenih — APPROVED** [review]

**Pull request merged**